### PR TITLE
 Replace sle-micro container with the one HPC builds for warewulf

### PR DESCRIPTION
**Pull request by b10n1k**
Use container image with the one provided by HPC and mentioned on the documentation

- Related ticket: https://progress.opensuse.org/issues/136355
- Verification run: 
https://aquarius.suse.cz/tests/19190
updated: https://aquarius.suse.cz/tests/19227
updated2: https://aquarius.suse.cz/tests/19235

### CODE REVIEW

**Review by schlad — CHANGES_REQUESTED** [review]
Please write better git commit message. 

To be more precise: apart from overall request to make the git commit message better, so write it using a bit better English, could you explain WHY you propose such change? The change itself if obvious so it is really required to explain  why you think changing the image is a good thing here.

File: tests/hpc/ww4.pm [hunk 1]
```diff
@@ -45,9 +45,9 @@ sub run ($self) {
     record_info "warewulf.conf", script_output("cat /etc/warewulf/warewulf.conf");
     record_info "defaults.conf", script_output("cat /etc/warewulf/defaults.conf");
 
-    $rt = (assert_script_run "wwctl container import docker://registry.suse.de/suse/containers/sle-micro/5.3/containers/suse/sle-micro-rancher/5.3:latest sle-micro-5.3 --setdefault", timeout => 320) ? 1 : 0;
+    $rt = (assert_script_run "wwctl container import docker://registry.suse.de/home/eeich/sle-hpc-15-sp5-container/containers/suse/sle-hpc-node:latest sle-hpc --setdefault", timeout => 320) ? 1 : 0;
```
Comment (schlad) [inline tests/hpc/ww4.pm:48]: oh gosh are we really gonna rely on home/eeich ?

I don't like such approach. Maybe to mitigate the bus factor, we could at least have a test variable for this?

Comment (czerw) [reply tests/hpc/ww4.pm:48]: wouldn't be some openSUSE based container better  or even smaller one like alpine linux ?

**Comment by czerw** [comment]
> Replace sle-micro container with the one HPC builds for warewulf

It is Warewulf (its name).

> Use container image with the one provided by HPC and mentioned on the documentation

In which documentation is it mentioned, could you please share the link?

File: tests/hpc/ww4.pm (hunk 1 above)
Comment (e4t) [reply tests/hpc/ww4.pm:48]: @schlad, a final solution is still WIP. I can try to create this in Devel:HPC - but the idea to have a variable is valid as the container is SP5 dependent. 
@czerw no. This is a *special* container meant for ww4 - it contains a kernel, systemd, identifies as SLE HPC and is built as a base container for compute node deployment. 
An alpine linux is *not* a special container for ww4.

**Comment by b10n1k** [comment]
> > Use container image with the one provided by HPC and mentioned on the documentation
> 
> In which documentation is it mentioned, could you please share the link?

i assume that should be https://gitlab.suse.de/HPC/warewulf-doc/-/blob/main/quickstart-sle.rst. But it is not updated yet

File: tests/hpc/ww4.pm [hunk 2]
```diff
@@ -45,9 +45,10 @@ sub run ($self) {
     record_info "warewulf.conf", script_output("cat /etc/warewulf/warewulf.conf");
     record_info "defaults.conf", script_output("cat /etc/warewulf/defaults.conf");
 
-    $rt = (assert_script_run "wwctl container import docker://registry.suse.de/suse/containers/sle-micro/5.3/containers/suse/sle-micro-rancher/5.3:latest sle-micro-5.3 --setdefault", timeout => 320) ? 1 : 0;
+    my $hpc_container = get_required_var('WAREWULF_CONTAINER');
```
Comment (czerw) [inline tests/hpc/ww4.pm:48]: Maybe `HPC_WAREWULF_CONTAINER` would be better, but current is still good.

File: tests/hpc/ww4.pm [hunk 3]
```diff
@@ -74,7 +75,7 @@ sub run ($self) {
         script_run("ssh -o StrictHostKeyChecking=accept-new $node ip a | tee /tmp/script_out");
         $rt = (assert_script_run "grep -E 'inet 192\.168\.10\.11[1-5]' /tmp/script_out", fail_message => 'IP address likely is not set or is not in the defined IP range!!') ? 1 : 0;
         test_case("Check IP on $node", 'Compute Validate IP', $rt);
-        $rt = validate_script_output("ssh -o StrictHostKeyChecking=accept-new $node cat /etc/os-release", sub { m/NAME.+SLE Micro/ });
+        $rt = validate_script_output("ssh -o StrictHostKeyChecking=accept-new $node cat /etc/os-release", sub { m/NAME.+SUSE Linux Enterprise High Performance Computing/ });
```
Comment (czerw) [inline tests/hpc/ww4.pm:78]: I think would good to also specify container name in variable `HPC_WAREWULF_CONTAINER_NAME` to have  it generic and ready for any container.

File: tests/hpc/ww4.pm [hunk 4]
```diff
@@ -45,9 +45,10 @@ sub run ($self) {
     record_info "warewulf.conf", script_output("cat /etc/warewulf/warewulf.conf");
     record_info "defaults.conf", script_output("cat /etc/warewulf/defaults.conf");
 
-    $rt = (assert_script_run "wwctl container import docker://registry.suse.de/suse/containers/sle-micro/5.3/containers/suse/sle-micro-rancher/5.3:latest sle-micro-5.3 --setdefault", timeout => 320) ? 1 : 0;
+    my $hpc_container = get_required_var('WAREWULF_CONTAINER');
+    $rt = (assert_script_run "wwctl container import docker://$hpc_container:latest sle-hpc --setdefault", timeout => 320) ? 1 : 0;
     test_case('Container pull', 'ww4', $rt);
-    $rt = (assert_script_run "wwctl profile set -y -C sle-micro-5.3") ? 1 : 0;
+    $rt = (assert_script_run "wwctl profile set -y -C sle-hpc") ? 1 : 0;
```
[the remark this answers is not in the document]
Comment (czerw) [reply tests/hpc/ww4.pm:51]: I would rather fill whole address into container to have full flexibility of container definition.

`wwctl container import $hpc_container:latest warewulf-container --setdefault`

File: tests/hpc/ww4.pm (hunk 2 above)
Comment (e4t) [reply tests/hpc/ww4.pm:48]: Namespacing is good, however, the risk of anything else acquiring the name 'warewulf' is probably low.

File: tests/hpc/ww4.pm (hunk 4 above)
[the remark this answers is not in the document]
Comment (czerw) [reply tests/hpc/ww4.pm]: > I would rather fill whole address into container variable to have full flexibility of container definition.
> 
> `wwctl container import $hpc_container warewulf-container --setdefault`

Any feedback or implementation about this one?

[the remark this answers is not in the document]
Comment (b10n1k) [reply tests/hpc/ww4.pm]: Done. https://aquarius.suse.cz/tests/19275

**Review by czerw — APPROVED** [review]

**Review by schlad — APPROVED** [review]

**Pull request merged**